source/MissionPanel.cpp

```diff
@@ -203,20 +203,22 @@ void MissionPanel::Draw()
 	MapPanel::Draw();
 
 	Color routeColor(.2f, .1f, 0.f, 0.f);
-	const System *system = selectedSystem;
-	while(distance.Days(system) > 0)
+	vector<const System*> plan = distance.Plan(selectedSystem);
+	const System* prev = player.GetSystem();
+	while(!plan.empty())
```

Comment:
Since we have a vector we can simply iterate backwards through the vector instead of doing this.